torch/csrc/lazy/core/ir_metadata.cpp

```diff
@@ -61,7 +73,10 @@ void PopScope() {
 }
 
 void ResetScopeContext() {
-  TORCH_CHECK(g_scope_context.scopes.size() == 0);
+  if (g_scope_context.scopes.size() != 0) {
```

Comment:
this is fine, but we could probably just use `TORCH_CHECK`?

`TORCH_CHECK(g_scope_context.scopes.size() == 0, "Expecting scope to be empty but it is " + GetCurrentScope())`